File: JavaScript/caesar/caesar.js
/**
 * Class CaesarCipher.
 * @author Christian M. Jensen <cmj@cjpg.dk>
 */
class CaesarCipher {
    /**
     * Initialise CaesarCipher class.
     * @param {int} key
     */
    constructor(key) {
        this._key = parseInt(key);
    }

    get key(){
        return this._key;
    }
    set key(key){
        this._key = parseInt(key);
    }

    /**
     * Decrypt a string using the Caesar cipher
     * @param {string} text
     * @returns {string}
     */
    decrypt(text) {
        let out = "";
        for (let i = 0; i < text.length; i++) {
            out += this.caesar(text[i], -this._key);
        }
        return out;
    }

    /**
     * Encrypt a string using the Caesar cipher
     * @param {string} text
     * @returns {string}
     */
    encrypt(text) {
        let out = "";
        for (let i = 0; i < text.length; i++) {
            out += this.caesar(text[i], this._key);
        }
        return out;
    }

    /**
     * get letter 'l' using caesar ecryption
     * @param {string} l the letter to ecrypt.
     * @param {int} k the key to use.
     * @returns {string}
     */
    caesar(l, k) {
        k = k % 26;
        let ch = this.shift(l);
        if(ch < 0 || ch > 25) {
            return l;
        }
        ch = ch + k;
        while (ch < 0)
        {
            ch = ch + 26;
        }
        while (ch > 25)
        {
            ch = ch - 26;
        }
        let a = this.isLower(l) ? "a".charCodeAt(0) : "A".charCodeAt(0);
        return String.fromCharCode(a + ch);
    }

    /**
     * shift the letter 'l' into a number between 0-25
     * @param {string} l the letter to shift.
     * @returns {number}
     */
    shift(l) {
        return l.toLowerCase().charCodeAt(0) - "a".charCodeAt(0);
    }

    /**
     * Check if str is all lowercase
     * @param {string} str the string to check
     * @returns {boolean} true: if all letters are lowercase.
     */
    isLower(str) {
        return str === str.toLowerCase();
    }
}

File: Python/caesar/caesar.py
# CS50x, 2019: Problem Set 6 (caesar)
from sys import argv


def main():
    # check arg lenght
    if len(argv) != 2:
        print(f"python {argv[0]} k")
        exit(1)

    # convert the key to int.
    key = int(argv[1])
    if not key > 0 and not key < 0:
        print(f"python {argv[0]} k")
        exit(1)

    # get plaintext from the use
    plaintext = input("plaintext : ")

    # print the ciphertext
    print("ciphertext: ", end="")
    for c in plaintext:
        if c.isalpha() and ord(c.lower()) in range(ord("a"), ord("z"), 1):
            print(caesar(c, key), end="")
        else:
            print(c, end="")
    print()


def shift(l):
    """shift the letter 'l' into a number between 0-25"""
    return ord(l)-ord("a") if l.islower() else ord(l)-ord("A");


def caesar(l, k):
    """get letter 'l' using caesar ecryption"""
    a = ord("a") if l.islower() else ord("A")
    return chr(a + (shift(l) + k) % 26)


if __name__ == "__main__":
    main()